core/src/manifest/native_transfer.rs

```diff
@@ -0,0 +1,159 @@
+use ethers::prelude::U64;
+use serde::{Deserialize, Deserializer, Serialize};
+
+use super::core::serialize_option_u64_as_string;
+use crate::manifest::{chat::ChatConfig, stream::StreamsConfig};
+
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+enum StringOrNum {
+    String(String),
+    Num(u64),
+}
+
+/// Deserialize a number or string into a U64. This is required for the untagged deserialize of
+/// native transfers to succeed.
+fn deserialize_option_u64_from_string_or_num<'de, D>(
+    deserializer: D,
+) -> Result<Option<U64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s: Option<StringOrNum> = Option::deserialize(deserializer)?;
+
+    match s {
+        Some(StringOrNum::String(string)) => {
+            U64::from_dec_str(&string).map(Some).map_err(serde::de::Error::custom)
+        }
+        Some(StringOrNum::Num(num)) => Ok(Some(U64::from(num))),
+        None => Ok(None),
+    }
+}
+
+#[derive(Debug, Hash, PartialEq, Eq, Serialize, Deserialize, Clone)]
+pub struct NativeTransferDetails {
+    pub network: String,
+
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_u64_from_string_or_num",
+        serialize_with = "serialize_option_u64_as_string"
+    )]
+    pub start_block: Option<U64>,
+
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "deserialize_option_u64_from_string_or_num",
+        serialize_with = "serialize_option_u64_as_string"
+    )]
+    pub end_block: Option<U64>,
+}
+
+fn default_enabled() -> bool {
+    true
+}
+
+/// The normalized 'Native Transfers' config.
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+pub struct NativeTransfers {
+    #[serde(default = "default_enabled")]
+    pub enabled: bool,
```

Comment:
dont think you need this anymore due to the enum type?

Reply:
It's a little easier to work with if I can coalesce the two types into a single shared struct, rather than discriminating on the enum in multiple places.

This allows the core field to be like this (rather than typing to make `native_transfers: false` coalesce to `None`:
```rs
#[serde(default)]
#[serde(deserialize_with = "deserialize_native_transfers")]
pub native_transfers: NativeTransfers,
```

Reply:
does this means it always has to have enabled: true in the struct itself in the yaml or is it just hidden away?

Reply:
It's completely hidden away, the struct should only have `enabled: false` if someone wanted to define everything, but turn it off.. But i didn't document it because it's not intended to be used.